app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -65,7 +65,13 @@ def build_enumerator(_run, cursor:)
       end
 
       @task.throttle_conditions.reduce(collection_enum) do |enum, condition|
-        enumerator_builder.build_throttle_enumerator(enum, **condition)
+        throttle_on, backoff = condition.values
+        backoff = backoff.call if backoff.respond_to?(:call)
+        enumerator_builder.build_throttle_enumerator(
+          enum,
+          throttle_on: throttle_on,
+          backoff: backoff
+        )
```

Comment:
Just a nitpick but perhaps it would be better to avoid relying on a particular values order just in case we may accidentally or intentionally change the structure here
https://github.com/Shopify/maintenance_tasks/blob/3a07296714c25043072b2c54702dfa16c38febd2/app/models/maintenance_tasks/task.rb#L109

```suggestion
        backoff = condition[:backoff]
        backoff = backoff.call if backoff.respond_to?(:call)
        enumerator_builder.build_throttle_enumerator(
          enum,
          throttle_on: condition[:throttle_on],
          backoff: backoff
        )
```

Reply:
Good point, done! ❤️